### ISSUE TITLE
Redirect the vocabulary to v2

### ISSUE THREAD

**Issue by iherman** [w3c/vc-data-model#966]
(The relevant file (ie, `.htaccess`) is not on the repository, so this cannot be submitted as a PR.)

PROPOSAL: change the redirections in the `.htaccess`, currently pointing at 

```
https://w3c.github.io/vc-data-model/vocab/credentials/credentials.{ttl,jsonld,html}
```

to

```
https://w3c.github.io/vc-data-model/vocab/credentials/v2/vocabulary.{ttl,jsonld,html}
```

Rationale. The v2 version of the vocabulary is backward compatible with the old one: all properties in v1 are retained in v2 (albeit, possibly, explicitly deprecated), so such a change would have no effect to deployed applications (which all use `https://www.w3.org/2018/credentials/`) for the vocabulary anyway. On the other hand, v2 is maintained, v1 is not...

**Comment by msporny** [comment]
+1, we should do this as soon as possible given that the new vocabulary is in better shape than the old one.

**Comment by TallTed** [comment]
+1

**Comment by msporny** [comment]
> @msporny @brentzundel @Sakurann : I am not sure what the decision procedure is for this issue; there is no PR to be made to go through our usual procedure. Who gives me the green light to proceed?

We could call for objections on the next call, and if there aren't any, implement the new redirect.

**Comment by iherman** [comment]
@msporny ping on this one?

**Comment by msporny** [comment]
The call right before thanksgiving went off of the rails. We didn't get around to calling for objections. You could just send a quick email to the mailing list to note that you're going to implement this by the end of this week and see if anyone disagrees. I doubt anyone would given that this is a fairly mundane update to make things better (have a better defined vocabulary).

**Comment by OR13** [comment]
I support using the list to ask for an objection and resolving this in 1 week, if there has been no objection.

**Comment by iherman** [comment]
The issue was discussed in a [meeting](https://www.w3.org/2017/vc/WG/Meetings/Minutes/2022-12-14-vcwg#section5-1) on 2022-12-14

- no resolutions were taken

<details><summary><i>View the transcript</i></summary>

#### 5.1. Redirect the vocabulary to v2 (issue vc-data-model#966)

_See github issue [vc-data-model#966](https://github.com/w3c/vc-data-model/issues/966)._




**Kristina Yasuda:** here's the topic to discuss - [https://github.com/w3c/vc-data-model/issues?q=is%3Aissue+is%3Aopen+label%3Adiscuss+sort%3Aupdated-asc](https://github.com/w3c/vc-data-model/issues?q=is%3Aissue+is%3Aopen+label%3Adiscuss+sort%3Aupdated-asc).  
… 966 - clarity sought on this.  

**Ivan Herman:** in an ideal world it would be a PR but it's a separate issue because the documents to be changed are not on the repository. The redirection from the official W3C address to Vocab for data model goes to the old v1 version of the vocab, and the proposal is to point it to version 2 which is maintained and in much better shape with V2 and backwards compatible. No downsides for doing it..  

**Kristina Yasuda:** proceed with this suggestion.  

**Manu Sporny:** strong support for doing this and moving it over..  

**Ivan Herman:** no PR just going to the W3C website and proceed with the suggestion!.  

> *Orie Steele:* Noting that this issue is related to the questions about the "core data model"... its weird to not here objections to it from the folks seeking to make context optional..

</details>

**Comment by selfissued** [comment]
I disagree with performing this redirection, as the V2 context may not stay compatible with the V1 context.  In particular, if `@vocab` is added to the V2 context, which it seems like we may do, we'd then have to revert the redirection.  Let's sort out the `@vocab` issues before doing this.  Thanks.

**Comment by dlongley** [comment]
@selfissued,

This issue, I think, is just about redirecting the vocabulary URL, not changing the context or doing something like redirecting the v1 context URL to the v2 context (which we should never do).

In other words, if someone were to follow any vocabulary URL (which is always optional to do) they would land on a page with a better description than what they land on now. This shouldn't be related to whether `@vocab` is used in a context or not -- or any other context changes -- as contexts only define mappings from JSON terms to globally unambiguous URLs, not what is at those URLs.

And, for the sake of clarity, no "JSON-LD processor" -- aka a library that implements the JSON-LD API -- ever follows those URLs, because it's not its job to do that. It only maps the JSON key "foo" to "https://example.com/foo" and only *if* you ask it to because you want to do something like create or verify a Data Integrity Proof.

There's no reason to use a JSON-LD library if you don't need to do that kind of transformation. If you just want to read the JSON after verifying a Data Integrity proof, you do not need a JSON-LD library (there is nothing for it to do). **I worry that some people may continue to mistakenly think the vocab URLs are fetched by JSON-LD libraries, but they are not.**

IMO, when working with VCs, a JSON-LD library is best thought of as operating at a similar layer that a JWT verification library operates -- it's used when checking a proof and then the verified JSON is passed off to another layer for use when it's done (if the proof verifies).

So, I don't think any decision about what is in the v2 context should affect this, but I may be missing something. If you have a different understanding, can you say more about what concrete technical problem you're worried about?

**Comment by OR13** [comment]
@selfissued we need to make these change regardless of if the v2 context includes vocab, since there are several related issues that are all blocked by this.

**Comment by OR13** [comment]
list of issues blocked by this issue:

- https://github.com/w3c/vc-data-model/issues/764
- https://github.com/w3c/vc-data-model/issues/810
- https://github.com/w3c/vc-data-model/issues/990

**Comment by kdenhartog** [comment]
Are we changing definitions of the terms in any way? If so we're effectively treating semantic drift as a feature not a bug here and I don't think that's wise if the whole point of signing the semantic data is to ensure that "we're talking about the same thing".

**Comment by iherman** [comment]
@kdenhartog There are no changes in moving from v1 to v2. V2 takes care of lots of mistakes, missing anchors, improved workflow to generate the files, etc.

Any semantic change should be addition of new terms, never removal, only declare a term as deprecated.

"we're talking about the same thing" is very much taken into account.

**Comment by iherman** [comment]
Redirection done.

@msporny @brentzundel or @Sakurann, I presume you have the right to close this issue.

**Comment by iherman** [comment]
> I disagree with performing this redirection, as the V2 context may not stay compatible with the V1 context. In particular, if `@vocab` is added to the V2 context, which it seems like we may do, we'd then have to revert the redirection. Let's sort out the `@vocab` issues before doing this. Thanks.

Beyond what @dlongley said.

I believe there is a misunderstanding. This issue _has nothing to do with the JSON-LD `@context`_. It is on the proper, RDF-compatible way of defining a _vocabulary_ (not a context) in various RDF syntaxes, in this case Turtle, JSON-LD, and HTML+RDFa. The various statements on the vocabulary (range and domain statements, for example) are only relevant for applications that rely on RDF reasoners that may, for example, detect some inconsistencies in the application data. It has, actually, nothing to do with JSON-LD (which only one of the several possible syntaxes for the same vocabulary).

**Comment by msporny** [comment]
This issue has been fixed, closing.
